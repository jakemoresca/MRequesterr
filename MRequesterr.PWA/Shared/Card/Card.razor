
    <div class="card shadow-sm text-white bg-dark" style="height: 290px">       
        @if (ImageUrl?.Length > 0)
        {
            <img src="@ImageUrl" class="card-img-top align-self-center" alt="@Title" style="width: auto; height: 290px">
        }
        else
        {
            <svg class="bd-placeholder-img card-img-top" width="100%" height="225" xmlns="http://www.w3.org/2000/svg"
                role="img" aria-label="Placeholder: Thumbnail" preserveAspectRatio="xMidYMid slice" focusable="false">
                <title>@Title</title>
                <rect width="100%" height="100%" fill="#55595c"></rect><text x="50%" y="50%" fill="#eceeef" dy=".3em">@Title</text>
            </svg>
        }

        <div class="card-body">
            <p class="card-text">@Title - @Overview</p>
            <p class="card-text">Released Status: @ReleasedStatus</p>
            <div class="d-flex justify-content-between align-items-center">
                <div class="btn-group">
                    <a class="btn btn-outline-primary" role="button">View</a>
                </div>
            </div>
        </div>
    </div>


@code {
    [Parameter]
    public string Title { get; set; }

    [Parameter]
    public string Overview { get; set; }

    [Parameter]
    public string ReleasedStatus { get; set; }

    [Parameter]
    public string? ImageUrl { get; set; }
}