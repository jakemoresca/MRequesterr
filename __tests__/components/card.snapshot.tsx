import { render } from '@testing-library/react'
import React from 'react';
import { RecoilRoot } from 'recoil';
import Card from '../../components/card';

it('renders card unchanged', () => {
  const statistics = {
    percentOfEpisodes: 100
  }

  const { container } = render(<RecoilRoot><Card title="title" isAvailable={true} statistics={statistics} hasFile={true} imageUrl="test" /></RecoilRoot>)
  expect(container).toMatchSnapshot()
})