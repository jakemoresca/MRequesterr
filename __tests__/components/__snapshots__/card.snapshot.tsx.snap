// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders card unchanged 1`] = `
<div>
  <div
    class="col bg-light card border-light"
  >
    <img
      alt="title"
      class="card-img-top"
      src="test"
      width="100%"
    />
    <div
      class="card-img-overlay"
    />
  </div>
</div>
`;
